test/net/unit/dhcp.cpp

```diff
@@ -19,34 +19,32 @@
 #include <net/inet>
 #include <net/interfaces>
 #include <hw/async_device.hpp>
+#include <net/dhcp/dhcpd.hpp>
 
-static std::unique_ptr<hw::Async_device<UserNet>> dev1 = nullptr;
```

Comment:
I didn't fully understand why these globals are not ok in unit tests anymore. Maybe something changed in how lest instantiates the unit tests 🤷 
Anyway, your solution also makes sure we don't have to depend on the test cases to be run in order for this to work, so it's better.

Reply:
I think it may just be that we create and store pointers to objects local to CASE() in the global static variables, and they then go out of scope (e.g. inet), but I'm not sure. It just seemed easier to put everything inside the functions to avoid that.